src/main/java/org/alfresco/rest/core/RestWrapper.java

```diff
@@ -255,7 +252,7 @@ public RestWrapper assertLastExceptionContains(String exception)
 
         T models = null;
 
-        if (!responseHasExceptions && !responseHasErrors)
+        if (!responseHasExceptions && !responseHasErrors && returnedResponse.getHeader("Content-Length") != null)
```

Comment:
In case of empty json array, content-length is always returned as null. EmptyJsonResponseException is returned when there is no response received from server but in this case there is a response from server but its an empty array.

Reply:
From what you said in Slack then I think this change is wrong. I think you should be changing this line:
```returnedResponse.jsonPath().getObject("list", classz);```
to also support:
```returnedResponse.jsonPath().getArray("list", classz);```